### PR TITLE
Escape quotation marks in text encoding

### PR DESCRIPTION
**Pull request by jRiest**
The switch statement [here](https://github.com/capnproto/go-capnproto2/blob/503c1bb7bbe95734920325b8999cc89b8b79fcfa/internal/strquote/strquote.go#L29-L34) is attempting to escape quotation marks, but since `needsEscape` didn't return `true` for quotation marks, they never made it into the switch statement, and therefore weren't getting escaped.

### CODE REVIEW

**Comment by jRiest** [comment]
Looks like the Bazel build is failing but the non-Bazel build passed, although that happened on the most recent commit to master too so I don't think it's caused by this change.

I'm not familiar with Bazel so I don't think I should be mucking about with the configuration, but based on these threads it sounds like it's an issue with go 1.13 and updating the versions of rules_go and gazelle might fix it.

https://github.com/golang/go/issues/33055
https://github.com/bazelbuild/bazel-gazelle/issues/639

File: encoding/text/marshal_test.go [hunk 1]
```diff
@@ -59,8 +59,8 @@ func TestEncode(t *testing.T) {
 		{0x81fdbfdc91779421, `(map = [])`, `(
     map = []
 )`},
-		{0x8e85252144f61858, `(data = "Hi\xde\xad\xbe\xef\xca\xfe")`, `(
-    data = "Hi\xde\xad\xbe\xef\xca\xfe"
+		{0x8e85252144f61858, `(data = "Hi\xde\xad\xbe\xef\xca\xfe\"\'\\")`, `(
```
Comment (lthibault) [inline encoding/text/marshal_test.go:62]: I'm okay with merging this optimistically, but could you just give me a bit of context about what's going on here?  The change seems very simple, but I'm not familiar with this part of the code base, so I just want to do my due diligence. 🙂

File: encoding/text/testdata/txt.capnp [hunk 2]
```diff
@@ -55,7 +55,7 @@ const mapVal @0xb167974479102805 :Value = (map = [
   (key = "foo", value = (void = void)),
   (key = "bar", value = (void = void)),
 ]);
-const data @0x8e85252144f61858 :Value = (data = 0x"4869 dead beef cafe");
+const data @0x8e85252144f61858 :Value = (data = 0x"4869 dead beef cafe 22275c");
```
Comment (lthibault) [inline encoding/text/testdata/txt.capnp:58]: Same here.  Can you just explain like I'm 5 what the `22275c` corresponds to and why it's needed?

**Review by lthibault — APPROVED** [review]
Changes are simple, so I'm okay with merging this even though I don't understand the details.  Could you just fill me in a little bit?

**Comment by lthibault** [comment]
@zenhack Does this seem OK to merge?

File: encoding/text/marshal_test.go (hunk 1 above)
Comment (zenhack) [reply encoding/text/marshal_test.go:62]: I assume this is just expanding the test, to cover the characters relevant to this patch.

File: encoding/text/testdata/txt.capnp (hunk 2 above)
Comment (zenhack) [reply encoding/text/testdata/txt.capnp:58]: This is just the hex encoding of the characters added above.

**Review by zenhack — APPROVED** [review]
LGTM.

**Pull request merged**